agents/locobot/label_prop.py

```diff
@@ -0,0 +1,348 @@
+        "items": list(cats), 
+    }
+    props.update(properties)
+    props_json = {
+        "items": list(props), 
+    }
+
+    # Write to file
+    with open(things_path, "w") as file: 
+        json.dump(cats_json, file)
+        print("saved categories to", things_path)
+    with open(props_path, "w") as file: 
+        json.dump(props_json, file)
+        print("saved properties to", props_path)
+
+def retrain_detector(settings): 
+
+    if len(settings) == 0: 
+        settings["trainSplit"] = 0.7
+        settings["learningRate"] = 0.005
+        settings["maxIters"] = 100
+
+    base_path = "annotation_data/"
+    coco_path = base_path + "coco/"
+    output_path = base_path + "output/"
+    model_path = base_path + "model/"
+    annotation_path = coco_path + "coco_results.json"
+    train_path = coco_path + "train.json"
+    test_path = coco_path + "test.json"
+    train_split = settings["trainSplit"]
+
+    # 1) Split coco json file into train and test using cocosplit code
+    # Adapted from https://github.com/akarazniewicz/cocosplit/blob/master/cocosplit.py
+    with open(annotation_path, "rt", encoding="UTF-8") as annotations_file: 
+
+        # Extract info from json
+        coco = json.load(annotations_file)
+        info = coco["info"]
+        licenses = coco["licenses"]
+        images = coco["images"]
+        annotations = coco["annotations"]
+        categories = coco["categories"]
+
+        # Remove images without annotations
+        images_with_annotations = set(map(lambda a: int(a["image_id"]), annotations))
+        images = list(filter(lambda i: i["id"] in images_with_annotations, images))
+
+        # Split images and annotations
+        x_images, y_images = train_test_split(images, train_size=train_split)
+        x_ids = list(map(lambda i: int(i["id"]), x_images))
+        x_annots = list(filter(lambda a: int(a["image_id"]) in x_ids, annotations))
+        y_ids = list(map(lambda i: int(i["id"]), y_images))
+        y_annots = list(filter(lambda a: int(a["image_id"]) in y_ids, annotations))
+
+        # Save to file
+        def save_coco(file, info, licenses, images, annotations, categories): 
+            with open(file, 'wt', encoding="UTF-8") as coco: 
+                json.dump({ 
+                    "info": info, 
+                    "licenses": licenses, 
+                    "images": images, 
+                    "annotations": annotations, 
+                    "categories": categories
+                }, coco, indent=2, sort_keys=True)
+        save_coco(train_path, info, licenses, x_images, x_annots, categories)
+        save_coco(test_path, info, licenses, y_images, y_annots, categories)
+
+    # 2) Use train/test files to retrain detector
+    # Adapted from train_detector.ipynb
+    dataset_name = "annotation_coco"
+    image_dir = base_path + "rgb/"
+    train_data = dataset_name + "_train"
+    test_data = dataset_name + "_test"
+
+    if train_data in DatasetCatalog.list(): 
```

Comment:
did `DatasetCatalog.clear()` and `MetadataCatalog.clear()` not work?

Reply:
I put these changes in the hollis/fix_retraining branch but I can include them here as well